### ISSUE TITLE
Link to Grafana Dashboards?

### ISSUE THREAD

**Issue by jason-riddle** [maxfields2000/dockerjenkins_tutorial#33]
Is there a link to the grafana dashboards at the bottom of https://engineering.riotgames.com/news/building-jenkins-inside-ephemeral-docker-container?

**Comment by maxfields2000** [comment]
Agh! No not yet, I really need to get that published. Will try to do it this weekend.

**Comment by jason-riddle** [comment]
Any update by chance?

**Comment by maxfields2000** [comment]
@jason-riddle 

It's not complete but here's a good start! https://github.com/maxfields2000/dockerjenkins_tutorial/tree/master/monitoring

**Comment by maxfields2000** [comment]
I still need to hook up cadvisor/container monitoring. Debating if I should just push up the influxdb0.9 I have or try to get t hings working on influxDB1.2.

**Comment by jason-riddle** [comment]
Awesome thanks! For the cadvisor monitoring stuff, maybe you should just release what you have and then make a note about it only working with influx 0.9 for now. Were there drastic changes between 0.9 and 1.2?

**Comment by maxfields2000** [comment]
That's probably what I'll do as it works and the images are still valid. Influx 1.2 no longer has an admin UI which I was using as part of my setup to create the right databases. Some API changes but I think Cadvisor has been updated to support it so I'll have to lock to an old version of cadvisor like I do at work.  Nothing to big.

So yea I'll release some cleanups with that in place with a big ol' caveat that a newer version is coming once I get my learn on with the new influx CLI/non-UI setup stuff.

**Comment by ghost** [comment]
Hello it seems like the graphite metrics report ignore pipeline jobs, only freestyle jobs seems to be reported in the dashboard. From my understanding, it's the metric jenkisn plugin that has issues with pipeline jobs be they single or multi branch

**Comment by ghost** [comment]
Also i tried the influxdb route, my issue with is that to get any data in influx, you basically have to add a post stage action to send job data to influxdb in every job... that method could be hard to scale

**Comment by maxfields2000** [comment]
@exuperok Oi, that's got to be because the "metrics" plugin in Jenkins doesn't recognize the pipeline project type, interesting I'll see what outstanding issues there are there or if it's reported int he data some other way.

**Comment by ghost** [comment]
@maxfields2000 I researching that too, for now I made a simple shared library so that developers adds a simple step like sendDatatoInflux to their pipeline as temp solution

**Comment by maxfields2000** [comment]
Ironically I did something similar, I wrote a shared pipeline library to extract that data by querying Jenkins via groovy and send it to Elastic Search (a breakdown of jobs by type actually, pipelines, folders, multi-branch, org, maven project, freestyle etc). It would appear that the metrics plugin just doesn't track anything except Project types which won't match the other job types because tehy don't inherit from it.

I can't really fix this here so I'm going to close this issue, but I'm noodling on how to do a good series on Jenkins metrics that will cover these challenges.
